### ISSUE TITLE
Getting error "relation "pgsodium.key_key_id_seq" does not exist (SQLSTATE 42P01)" when using "db remote commit"

### ISSUE THREAD

**Issue by xsmithdev** [supabase/supabase#8616]
# Bug report

## Describe the bug

When using the latest CLI to init, start, link, and remote commit; supabase errors with a missing relation.

```bash
relation "pgsodium.key_key_id_seq" does not exist (SQLSTATE 42P01)
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. `supabase init`
2. `supabase start`
3. `supabase link --project ...`
4. `supabase db remote commit`
5. `supabase db reset`

```bash
PS C:\Users\silen\Documents\GitHub\buildr-monorepo> supabase db reset        
Resetting database...
Initialising schema...
Applying migration 20220824061131_remote_commit.sql...
Error: ERROR: relation "pgsodium.key_key_id_seq" does not exist (SQLSTATE 42P01)
```

## Expected behavior

The database should now reflect the migration.

## System information

- OS: Windows 11
- Version of supabase-cli: 1.1.5
- Version of Node.js: 16.13.2

**Comment by ZetiMente** [comment]
gah, I just got the same error, did u make any progress on this?

**Comment by alemontesdev** [comment]
Hello, for me now it worked just deleting the migration file that I just created for the very first commit. But still I don't know why is it happening.

**Comment by xsmithdev** [comment]
No progress. I still don't understand why this happens. It's very annoying and I'm completely unable to sync my local to my remote.

**Comment by magicseth** [comment]
I'm hitting this as well

**Comment by ARMATAV** [comment]
Same here

**Comment by ARMATAV** [comment]
Fixed it;

https://github.com/supabase/cli/issues/246

1. `supabase link --project-ref <project-id>`
2. `supabase db remote commit`
3. `supabase db reset`
4. Cool, now we can see that we're screwed and we get extension errors like `pgsodium.key_key_id_seq does not exist`
5. Great, go into the `20220913230812_remote_commit.sql` file - comment everything out, save without formatting
6. Create a new `.sql` file over at `supabase/extensions.sql` and put in whichever extension is failing you `CREATE EXTENSION pgsodium SCHEMA extensions;` in this case
7. Great, `supabase db reset` and it should work
8. However, we're still screwed because when we `supabase db push` to the remote, you'll see the timestamp is different
9. Go into your production database (or the remote as I called it), and go to `supabase_migrations.schema_migrations` - copy the one row in there (if you have multiple, it should be the first)
<img width="187" alt="image" src="https://user-images.githubusercontent.com/2433203/190028614-070fb4d1-50d0-4aba-ac06-2e5e9bc9ce4b.png">
10. Paste that over your original timestamp in your `migrations` folder
<img width="333" alt="image" src="https://user-images.githubusercontent.com/2433203/190028656-195fe4e1-e402-4965-be99-d3108876df00.png">
11. `supabase db push`

And you're good to go. Verified that this works even when I add new tables and `supabase db reset`.

Hope this saves someone the absolutely agonizing experience of googling "relation "pgsodium.key_key_id_seq" does not exist (SQLSTATE 42P01)"

**Comment by ZetiMente** [comment]
Hi @ARMATAV I was able to get to step 10. Thank you !!! However, I don't want to do a `supabase db push`, I'm looking to just have my local db in sync with the remote. I don't want to push my local to my remote. However, when I do `supabase db reset` again, I ran into the error again once I uncomment the migration code.

**Comment by ARMATAV** [comment]
Hard to say @ZetiMente - I think you can just leave `remote_commit` commented out. 

Also - are you using their latest CLI version?

**Comment by ZetiMente** [comment]
@ARMATAV Yes, I am on 1.4.6. Not sure what you mean by "leave remote_commit" commented out? If I leave the entire migration commented out, then how do I sync my local db to it?

**Comment by ARMATAV** [comment]
@ZetiMente Over here https://github.com/supabase/cli/issues/246#issuecomment-1092879386
